tests/unit/types/arrays/test_neural_ops.py

```diff
@@ -52,19 +131,22 @@ def docarrays_for_embedding_distance_computation_sparse():
     return D1, D2
 
 
+@pytest.fixture
+def embeddings():
+    return np.array([[1, 0, 0], [2, 0, 0], [3, 0, 0]])
+
+
 @pytest.mark.parametrize(
     'limit, batch_size', [(1, None), (2, None), (None, None), (1, 1), (1, 2), (2, 1)]
 )
-def test_matching_retrieves_correct_number(
-    docarrays_for_embedding_distance_computation, limit, batch_size
-):
-    D1, D2 = docarrays_for_embedding_distance_computation
-    D1.match(D2, metric='sqeuclidean', limit=limit, batch_size=batch_size)
-    for m in D1.get_attributes('matches'):
-        if limit is None:
-            assert len(m) == len(D2)
-        else:
-            assert len(m) == limit
+def test_matching_retrieves_correct_number(docarray_combinations, limit, batch_size):
```

Comment:
this `docarray_combinations` loop should not be done by us, but this should be part of the `pytest.mark.parametrize`

Reply:
Otherwise we cannot get a report on which combination failed the test.

Reply:
fixed